src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -131,17 +156,46 @@ public void configure(Map<String, ?> configs) {
         storageQuotaHard = config.getHardStorageQuota();
         storageCheckInterval = config.getStorageCheckInterval();
         logDirs = Arrays.stream(config.getLogDirs().split(",")).map(Paths::get).collect(Collectors.toList());
+        thresholdMap = new HashMap<>();
+        String percents = config.getStoragePercents();
+        String[] split = percents.split("\\|");
+        for (String s : split) {
+            String[] inner = s.split(":");
+            Path path = Paths.get(inner[0]);
+            if (!logDirs.contains(path)) {
+                throw new IllegalArgumentException(StaticQuotaConfig.STORAGE_QUOTA_PERCENTAGES_PROP + " property contains path that's not part of logDirs: " + path + " - " + logDirs);
+            }
+            double soft = Double.parseDouble(inner[1]);
+            double hard = Double.parseDouble(inner[2]);
+            FileStore store = apply(() -> Files.getFileStore(path));
```

Comment:
a try/catch woudl be clearer here than `apply`, which was only added to facilitate use with the Java `Stream` API.

Reply:
Do we really need to bother with this?
Afais `apply` was added to make this pesky IOException go away.
And I use it here for exactly that purpose.

Reply:
I guess if we're keeping `apply` we can use it here.